1/ToDoApp/src/components/ToDoItem/ToDoItem.spec.js

```diff
@@ -0,0 +1,46 @@
+import { mount } from '@vue/test-utils'
+import ToDoItem from './ToDoItem.vue'
+
+describe('ToDoItem', () => {
+  describe('given a todo', () => {
+    const wrapper = mount(ToDoItem, {
+      propsData: {
+        todo:{
+          text: 'test-todo'
+        }
+      }
+    })
+    it('renders todo.text', () => {
+      expect(wrapper.vm.todo.text).toBe('test-todo');
+    })
+    it('does not show input field', () => {
+      expect(wrapper.vm.editModeOn).toBe(false);
+    })
+    describe('click on Edit Button', () => {
+      const editButton=wrapper.find('#editButton');
+      it('shows input field', () => {
+        expect(wrapper.find('#editButton').exists()).toBe(true);
+        editButton.trigger('click');
+        expect(wrapper.vm.editModeOn).toBe(true);
```

Comment:
Don't expect sth. on `wrapper.vm`. Only as a last resort.

The `wrapper`'s html should be it's interface. So, when it's in edit mode, I believe some button or input field will be visible, no?

If you don't treat your components as a black box, you will have a hard time to fix the tests for false positives when you do a refactoring of the implementation of the component.